### PR TITLE
15.1 docs: `unstablePersistentCaching`

### PR DESCRIPTION
**Pull request by delbaoliveira**
**DO NOT MERGE.**

Closes: https://linear.app/vercel/issue/DOC-3817/[experimental]-turbopack-incremental-caching

Also: 
- Adds missing `turbo` options
- Adds missing types

Related PRs: 
- https://github.com/vercel/next.js/pull/70904
- https://github.com/vercel/next.js/pull/71090

### CODE REVIEW

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 1]
```diff
@@ -45,14 +45,18 @@ module.exports = nextConfig
...
+| `minify`                  | `boolean`                                       | Enable minification. Defaults to true in build mode and false in dev mode. |
+| `resolveAlias`            | `Record<string>`                                | Map aliased imports to modules to load in their place.                     |
+| `resolveExtensions`       | `string[]`                                      | List of extensions to resolve when importing files.                        |
+| `rules`                   | `Record<string, TurboRuleConfigItemOrShortcut>` | List of unsupported webpack loaders to apply when running with Turbopack.  |
+| `root`                    | `string`                                        | The repo root directory.                                                   |
+| `treeShaking`             | `boolean`                                       | Enable tree shaking for the turbopack dev server and build.                |
+| `unstablePersistentCache` | `boolean`                                       | Enable the persistent cache for turbopack dev and build servers.           |
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:59]:
```suggestion
| `unstablePersistentCaching` | `boolean`                                       | Enable the persistent cache for Turbopack dev server and build.           |
```

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 2]
```diff
@@ -45,14 +45,18 @@ module.exports = nextConfig
...
-| ------------------- | ------------------------------------------------------------------------- |
-| `rules`             | List of unsupported webpack loaders to apply when running with Turbopack. |
-| `resolveAlias`      | Map aliased imports to modules to load in their place.                    |
-| `resolveExtensions` | List of extensions to resolve when importing files.                       |
-| `moduleIdStrategy`  | Assign module IDs                                                         |
-| `treeShaking`       | Enable tree shaking for the turbopack dev server and build.               |
-| `memoryLimit`       | A target memory limit for turbo, in bytes.                                |
+| Option                    | Type                                            | Description                                                                |
+| ------------------------- | ----------------------------------------------- | -------------------------------------------------------------------------- |
+| `loaders`                 | `Record<string, TurboLoaderItem[]>`             | List of webpack loaders to apply when running with Turbopack.              |
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:50]: That doesn't exist anymore. It has been removed in favor of `rules`

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 3]
```diff
@@ -45,14 +45,18 @@ module.exports = nextConfig
...
+| `memoryLimit`             | `number`                                        | A target memory limit for turbo, in bytes.                                 |
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:51]: This shouldn't be documented for now...

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 4]
```diff
@@ -68,6 +72,36 @@ The following loaders have been tested to work with Turbopack's webpack loader i
...
+
+To enable persistent caching, set the `unstablePersistentCache` option to `true` in `next.config.js`:
+
+```ts filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  experimental: {
+    turbo: {
+      unstablePersistentCache: true,
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:87]:
```suggestion
      unstablePersistentCaching: true,
```

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 5]
```diff
@@ -68,6 +72,36 @@ The following loaders have been tested to work with Turbopack's webpack loader i
...
+}
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+module.exports = {
+  experimental: {
+    turbo: {
+      unstablePersistentCache: true,
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:99]:
```suggestion
      unstablePersistentCaching: true,
```

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx [hunk 6]
```diff
@@ -68,6 +72,36 @@ The following loaders have been tested to work with Turbopack's webpack loader i
 
 ## Examples
 
+### Enabling Persistent Caching
+
+Turbopack's Persistent Caching allows compilation work to persist across restarts of the development server or builds, reducing cold start times for all applications by caching previously compiled work, and only recompiling what has changed.
+
+To enable persistent caching, set the `unstablePersistentCache` option to `true` in `next.config.js`:
```
Comment (sokra) [inline docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:79]:
```suggestion
To enable persistent caching, set the `unstablePersistentCaching` option to `true` in `next.config.js`:
```

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx (hunk 3 above)
Comment (delbaoliveira) [reply docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:51]: Removed 👍🏼

File: docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx (hunk 2 above)
Comment (delbaoliveira) [reply docs/01-app/03-api-reference/05-config/01-next-config-js/turbo.mdx:50]: Removed 👍🏼

**Comment by delbaoliveira** [comment]
Closing as this feature has been moved to 15.2

**Pull request closed without merging**